encodings/dict/benches/dict_mask.rs

```diff
@@ -0,0 +1,59 @@
+#![allow(clippy::unwrap_used)]
+
+use divan::Bencher;
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng as _};
+use vortex_array::array::PrimitiveArray;
+use vortex_array::compute::mask;
+use vortex_array::IntoArray as _;
+use vortex_dict::DictArray;
+use vortex_mask::Mask;
+
+fn main() {
+    divan::main();
+}
+
+fn filter_mask(len: usize, fraction_masked: f64, rng: &mut StdRng) -> Mask {
+    let indices = (0..len)
+        .filter(|_| rng.gen_bool(fraction_masked))
+        .collect::<Vec<usize>>();
+    Mask::from_indices(len, indices)
+}
+
+#[divan::bench(args = [
+    (0.9, 0.9),
+    (0.9, 0.5),
+    (0.9, 0.1),
+    (0.9, 0.01),
+    (0.5, 0.9),
+    (0.5, 0.5),
+    (0.5, 0.1),
+    (0.5, 0.01),
+    (0.1, 0.9),
+    (0.1, 0.5),
+    (0.1, 0.1),
+    (0.1, 0.01),
+    (0.01, 0.9),
+    (0.01, 0.5),
+    (0.01, 0.1),
+    (0.01, 0.01),
+])]
+fn bench_dict_mask(bencher: Bencher, (fraction_valid, fraction_masked): (f64, f64)) {
```

Comment:
This is a useful microbenchmark but dict currently lacks a MaskFn so this just measures decompression time. I intend to follow up with a real MaskFn.